packages/shortcode/src/types.ts

```diff
@@ -0,0 +1,210 @@
+/**
+ * Shortcode attributes object.
+ */
+export type ShortcodeAttrs = {
+	/**
+	 * Object with named attributes.
+	 */
+	named: Record< string, string | undefined >;
+
+	/**
+	 * Array with numeric attributes.
+	 */
+	numeric: string[];
+};
+
+export type ShortcodeMatch = {
+	/**
+	 * Index the shortcode is found at.
+	 */
+	index: number;
+
+	/**
+	 * Matched content.
+	 */
+	content: string;
+
+	/**
+	 * Shortcode instance of the match.
+	 */
+	shortcode: Shortcode;
+};
+
+/**
+ * Shortcode options.
+ */
+export interface ShortcodeOptions {
+	/**
+	 * Shortcode tag.
+	 */
+	tag: string;
+
+	/**
+	 * Shortcode attributes.
+	 */
+	attrs?: Partial< ShortcodeAttrs > | string;
+
+	/**
+	 * Shortcode content.
+	 */
+	content?: string;
+
+	/**
+	 * Shortcode type: `self-closing`, `closed`, or `single`.
+	 */
+	type?: 'self-closing' | 'closed' | 'single';
+}
+
+/**
+ * Shortcode object.
+ */
+export interface Shortcode extends ShortcodeOptions {
+	/**
+	 * Shortcode attributes.
+	 */
+	attrs: ShortcodeAttrs;
+}
+
+export type Match =
+	| NonNullable< ReturnType< RegExp[ 'exec' ] > >
+	| Array< string >;
+
+export type ReplaceCallback = ( shortcode: Shortcode ) => string;
+
+/**
+ * WordPress Shortcode instance.
+ */
+export interface WPShortcode {
```

Comment:
Can we drop somehow this WP prefix? It doesn’t add any value in TypeScripts as these names aren’t global like in JSDoc. It also differs on DefinitelyTyped:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c39e00d70995131a814a23b8054ea191d910925b/types/wordpress__shortcode/index.d.ts#L116

Reply:
`Shortcode` is already used above, that's why I used that. We can get rid of it once we convert that file to TS.

Reply:
We only need to do it in one release cycle so folks don't have to change these types after the get published to npm 👍

Reply:
Alright, I have removed the `WP` prefix. Thanks